base/range.jl

```diff
@@ -860,13 +860,18 @@ end
 
 median(r::Range{<:Real}) = mean(r)
 
-function in(x, r::Range)
+function _in_range(x, r::Range)
     n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
```

Comment:
> Maybe compute the second operand unconditionally and call one on that?

I imagine the branch exists to avoid the potential division by zero (`step(r)`) in computing the second operand (and potential downstream issues)?

Perhaps continuing to branch on `step(r) == 0`, but eliminating `n` in the `step(r) == 0` case would work well? For example, perhaps something along the lines of
```julia
function _in_range(x, r::Range)
    if step(r) == 0
        isempty(r) ? false : first(r) == x
    else
        n = round(Integer, (x - first(r)) / step(r)) + 1
        n >= 1 && n <= length(r) && r[n] == x
    end
end
```
Thoughts?